README.md

```diff
@@ -10,4 +10,396 @@ TODO: description
+- `title` _{string}_ - The medical contact's title
+- `first_name` _{string}_ - The medical contact's first name
+- `last_name` _{string}_ - The medical contact's last name
+- `hospital` _{string}_ - The medical contact's associated hospital
+- `specialty` _{string}_ - The medical contact's specialty
+- `phone_number` _{string}_ - The medical contact's phone number
+- `notes` _{string}_ - Notes about the medical contact
+
+**Returns**
+
+- A success message
+- An object with the edited medical contact
+
+**Throws**
+
+- `403` if the user is not logged in
+- `400` If the first_name, last_name, or phone_number is empty or a stream of empty spaces
+
+#### `DELETE /api/medicalContacts/:medicalContactId?` - Delete a medical contact
+
+**Returns**
+
+- A success message
+
+**Throws**
+
+- `403` if the user is not logged in
+- `403` if the user is not the owner of the medical contacts
+- `404` if the medicalContactId is invalid
+
+### Medication
+
+#### `GET /api/medications?owner=USERNAME` - Get medications by owner
+
+**Returns**
+
+- An array of medications created by user with username `owner` in alphabetical order with respect to the name of the medication
+
+**Throws**
+
+- `400` if `owner` is not given
+- `404` if `owner` is not a recognized username of any user
+
+#### `GET /api/medications?name=NAME` - Get medications by name
+
+**Returns**
+
+- An array of medications with name `NAME` in alphabetical order with respect to the name of the medication
+
+**Throws**
+
+- `400` if `NAME` is not given
+
+#### `POST /api/medications` - Create a new medication
+
+**Body**
+
+- `active` _{string}_ - Whether the medication is in use or not
+- `name` _{string}_ - The medication's name
+- `generic_name` _{string}_ - The medication's generic compound name
+- `dose` _{string}_ - The medication's dose
+- `notes` _{string}_ - Notes about the medication
+
+**Returns**
+
+- A success message
+- An object with the created medication
+
+**Throws**
+
+- `403` if the user is not logged in
+- `400` If the name, generic_name, or dose is empty or a stream of empty spaces
+
+#### `PATCH /api/medications/:medicationId?` - Edit the information of a medication
+
+**Body**
+
+- `active` _{string}_ - Whether the medication is in use or not
+- `name` _{string}_ - The medication's name
+- `generic_name` _{string}_ - The medication's generic compound name
+- `dose` _{string}_ - The medication's dose
+- `notes` _{string}_ - Notes about the medication
+
+**Returns**
+
+- A success message
+- An object with the edited medication
+
+**Throws**
+
+- `403` if the user is not logged in
+- `400` If the name, generic_name, or dose is empty or a stream of empty spaces
+
+#### `DELETE /api/medications/:medicationId?` - Delete a medication
+
+**Returns**
+
+- A success message
+
+**Throws**
+
+- `403` if the user is not logged in
+- `403` if the user is not the owner of the medication
+- `404` if the medicationId is invalid
+
+### Entry
+
+TODO: Linda
+
+
+### Trend
```

Comment:
here's what im thinking: 
a trend is a collection of points. in mongodb, it can be stored as {_id, title, values}, where the title of the displayed graph === the title from the log, and the values is an array of integers === the condition scale values from each of the user's log entries with the same titles

Reply:
This makes sense I think. In the concept design we used "label" instead of "title" but if you prefer title I think that's okay too.

One important thing that you are missing here is that each point in the trend needs to store the item (i.e. log entry) it is associated with as well as its date/timestamp. If you only have the y values (i.e. no dates) you won't be able to graph anything, and we won't be able to do syncs down the line if we don't have the reference to the item.

Reply:
Ok I had left some more detailed comments in the code, but I just deleted them because I think there is a bigger concern that needs to be addressed first. If you refer back to the concept design, I don't think that creation or deletion of a trend are actions that are part of this concept. In the beta design, these are hardcoded (so you should just check that the title/label is valid when adding a point), and in the final design the adding/removing of labels is actually the whole point of the condition concept rather than part of this concept.

I would recommend rewriting your routes to be in terms of creating/updating/deleting points instead. How you choose to model that in Mongo (indidvidual point documents or add them to arrays based on label) is an implementation detail that you can choose later and doesn't need to be part of the external interface.

Reply:
Separate question for everyone: do we want log entry to require two sequential requests (first use entry api to create entry, then use trend api to add data point) or a single request that makes two backend calls? Personally I prefer the latter, but I see from a concept independence (and perhaps also work independence) standpoint you could argue for the former.

The way I treated concept indepenedence in A5 was that the /router.ts files were fair game to call any collection or middleware methods but all of the other files should generally be restricted to only interacting with their own concept. How did others interpret this?

Reply:
to answer the latter of these two comments, I think keeping points and entry as two api calls could be the best, as I think it would most easily allow extra feature development in the future. in addition, this "two-call" logic is what I tried to build my trend api routes off of.